samoa-instances/src/main/java/org/apache/samoa/instances/AvroLoader.java

```diff
@@ -0,0 +1,285 @@
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.avro.Schema;
+import org.apache.avro.Schema.Field;
+import org.apache.avro.generic.GenericData.EnumSymbol;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+
+/**
+ * Load Data from Avro Stream and parse to corresponding Dense & Parse Instances
+ * Abstract Class: Subclass this class for different types of Avro Encodings
+ * 
+ *
+ */
+public abstract class AvroLoader implements Loader {
+
+	private static final long serialVersionUID = 1L;
+
+	/** Representation of the Avro Schema for the Instances being read. Built from the first line in the data  */
+	protected Schema schema = null;
+
+	/**  Meta-data of the Instance */
+	protected InstanceInformation instanceInformation;
+
+	/** List of attributes in the data as read from the schema */
+	protected List<Attribute> attributes;
+
+	/** This variable is to check if the data stored is Sparse or Dense */
+	protected boolean isSparseData;
+
+	protected int classAttribute;
+
+	/** Datum Reader for Avro Data*/
+	public DatumReader<GenericRecord> datumReader = null;
+
+	public AvroLoader(int classAttribute) {
+		this.classAttribute = classAttribute;
+		this.isSparseData = false;
+	}
+
+	/** Intialize Avro Schema, Meta Data, InstanceInformation from Input Avro Stream */
+	public abstract void initializeSchema(InputStream inputStream);
+
+	/** Read a single SAMOA Instance from Input Avro Stream */
+	public abstract Instance readInstance();
+	
+	/**
+	 * Method to read Dense Instances from Avro File
+	 * @return Instance
+	 */
+	protected Instance readInstanceDense(GenericRecord record)
+	{
+		Instance instance = new DenseInstance(this.instanceInformation.numAttributes() + 1);
+		int numAttribute = 0;
+
+		for (Attribute attribute : attributes) {
+			Object value = record.get(attribute.name);
+
+			boolean isNumeric = attributes.get(numAttribute).isNumeric();
+			boolean isNominal = attributes.get(numAttribute).isNominal();
+
+			if(isNumeric)
+			{
+				if(value instanceof Double)	
+					this.setDenseValue(instance, numAttribute, (double)value);
+				else if (value instanceof Long)	
+					this.setDenseValue(instance, numAttribute, (long)value);
+				else if (value instanceof Integer)	
+					this.setDenseValue(instance, numAttribute, (int)value);
```

Comment:
The avro record.get(attribute.name) would give an instance of java.lang.Integer or java.lang.Double or java.lang.Long for numeric type . The ** if(value instanceof java.lang.Integer ) ** on a java.lang.Double object will never give us true. Same case vice versa too since they don't have a direct inheritance relationship. So in my opinion, it wouldn't matter which ever order we write. Is there any other particular reason you think the order need to be changed ? Perhaps users are more likely to go with Integer in the schema ??